### PR TITLE
Update help pages

### PR DESCRIPTION
**Pull request by humdingerb**
Shoehorn in the additional feature of translating Java properties
and line-separated files.

Rename "MIME type" to "App signature", which I feel is more
commonly used (see makefile_engine)

Still missing: What's the "Last key empty"? See the ???? placeholder in the developer.twig.

### CODE REVIEW

File: resources/views/help/developer.twig [hunk 1]
```diff
@@ -38,66 +38,94 @@ Strings that aren't in the new <i>en.catkeys</i> anymore get removed (together w
 
 <p>When all the languages you need are 100% finished, you can export and download their <i>xx.catkeys</i> to use when building your application.</p>
 
+<h2>Non-catkeys translations</h3>
+
+<p>Polyglot can be used for the translation of other files than Haiku catkeys:</p>
+<ul>
+<li><p><b>Line separated files</b> are any text files where the to-be-translated text blocks are separated by lines with certain characters (i.e. a <tt>#</tt> or just an empty line. As an example, see an export of the tips files of <i>Tipster</i> here at Polyglot. ReadMe texts and other documentation are other candidates for this type of file.</p></li>
```
Comment (KapiX) [inline resources/views/help/developer.twig:45]: missing ) after "e.g. #"

File: resources/views/help/developer.twig [hunk 2]
```diff
@@ -38,66 +38,94 @@ Strings that aren't in the new <i>en.catkeys</i> anymore get removed (together w
...
+
+<p>Here you enter the name of the translation module you want to add. If it's for the main part of your app it should be called simply "Application". Then you choose the kind of file (Haiku catkeys, line-separated, or Java properties) from the pop-up menu and click on "Add".<p>
 
-<p>After adding the translation module for this main part of your app (by default named "Application"), you upload the <i>en.catkeys</i> for it:</p>
+<p>You arrive at a page to upload the <i>English original</i> for the added module (here for a catkeys module):</p>
 
 <p>{{ html_image('images/help/dev-upload-en-catkeys.png', 'Uploading the en.catkeys') }}</p>
 
-<p>Below that you'll see the name of the translation module, the signature of the application and the checksum which changes every time you create a new <i>en.catkeys</i> with changed GUI strings:</p>
+<p>After you've uploaded a file, you'll see for all kind of files:</p>
```
Comment (KapiX) [inline resources/views/help/developer.twig:64]: kinds

File: resources/views/help/developer.twig [hunk 3]
```diff
@@ -38,66 +38,94 @@ Strings that aren't in the new <i>en.catkeys</i> anymore get removed (together w
...
+<li><i>Checksum</i> - the checksum which changes every time you create a new <i>en.catkeys</i> with changed GUI strings.</li>
+</ul>
 
-<p>{{ html_image('images/help/dev-name-sig-checksum.png', 'The data of a translation module') }}</p>
+<p>Additionally for line-separated files:</p>
+<p>{{ html_image('images/help/dev-upload-line-separated.png', 'Uploading the line-separated file') }}</p>
+<ul>
+<li><i>Extension</i> - the file extension of the uploaded file.</li>
+<li><i>Separator</i> - the characters used to separate translation blocks. Leave empty to use an empty line.</li>
+<li><i>Last key empty</i> - ??????????</li>
```
Comment (KapiX) [inline resources/views/help/developer.twig:82]: This is internal variable, it keeps track of last line break (or separator) - in Tipster .txt is ended with "%" and no text after - parser reads that as empty key.

File: resources/views/help/developer.twig [hunk 4]
```diff
@@ -38,66 +38,94 @@ Strings that aren't in the new <i>en.catkeys</i> anymore get removed (together w
...
+</ul>
 
-<p>Under that follows the contents of your <i>en.catkeys</i>.</p>
+<p>Under all that follows the contents of your uploaded file.</p>
 
-<p>If your project only uses a single <i>en.catkeys</i> file that contains all the strings, you're done now. If you use several <i>en.catkeys</i> to build your software, maybe for a "Preferences panel", a "Tracker-addon, and a "Library", you need several "Translation modules" and need to repeat these steps and add the appropriate identifying names and upload their <i>en.catkeys</i>.</p>
+<p>If your project only uses a single <i>en.catkeys</i> file that contains all the strings, you're done now. If you use several <i>en.catkeys</i> to build your software, maybe for a "Preferences panel", a "Tracker-addon, and a "Library", you need several "Translation modules" and need to repeat these steps and add the appropriate identifying names and upload their <i>en.catkeys</i> (or line-separated files, or Java properties).</p>
```
Comment (KapiX) [inline resources/views/help/developer.twig:87]: Missing " after "Tracker add-on"

**Comment by humdingerb** [comment]
Cannot reply to a specific comment in Haiku browsers... :(

WRT "Last key empty", does this internal variable need to be exposed in the interface?

**Comment by KapiX** [comment]
> WRT "Last key empty", does this internal variable need to be exposed in the interface?

No, I was just lazy :) I will hide them sometime soon...

I think you need to rebase your branch because there are conflicts.

**Pull request closed without merging**